src/extension.ts

```diff
@@ -105,6 +105,25 @@ export function activate(context: ExtensionContext) {
 							lastStatus = item.text;
 							resolve();
 							commands.registerCommand(Commands.START_DEBUG_SESSION, async (config) => {
+								if (!config.request) { // if 'request' is missing interpret this as a missing launch.json
+									config.type = 'java';
+									config.name = 'Java Debug';
+									config.request = 'launch';
+									const editor = vscode.window.activeTextEditor;
+									if (editor && editor.document.languageId === 'java') {
+										const fullpath = editor.document.fileName;
+										const cur = path.parse(fullpath).dir;
+										config.startupClass = path.parse(fullpath).base.slice(0, -5);
+										config.sourcePath = [cur];
```

Comment:
for single file debug,  the file  is linked to default project under src/[package]. so if we support non-default package, we also need to support user to change package name, i don't think we can do that as we cannot create working copy for .project file